teps/0056-pipelines-in-pipelines.md

```diff
@@ -352,6 +359,225 @@ main-`PipelineRun` to be able to fetch and access the sub-`PipelineRun`'s full s
 6. Users should be able to propagate actions from the main-`Pipeline` to the sub-`Pipeline`, such as deletion and
 cancellation.
 
+## Proposal
+
+This proposal focuses on enabling specification and execution of a `Pipeline` in a `Pipeline`. This section will 
+provide the overview, see the [design](#design) section below for further details.
+
+### Specification
+
+To support defining and executing `Pipelines` in `Pipelines`, we propose adding a `Pipeline` field in `PipelineTask` 
+which takes the [`PipelineRunSpec`][pipelinerunspec]. This field should take `PipelineRef` and `PipelineSpec` only 
+initially, we will add validation to confirm this. This field provides the extensibility to support other fields
+related to creating a `PipelineRun` from [`PipelineRunSpec`][pipelinerunspec]. We will create 
+
+```go
+type PipelineTask struct {
+    Name     string                `json:"name,omitempty"`
+    TaskRef  *TaskRef              `json:"taskRef,omitempty"`
+    TaskSpec *EmbeddedTask         `json:"taskSpec,omitempty"`
+    Pipeline *PipelineRunSpec      `json:"pipeline,omitempty"`
+    ...
+}
+```
+
+For example, the [example](#fanning-out-pipelines) described above can be solved using a `Pipeline` named 
+`security-scans` which is run within a `Pipeline` named `clone-scan-notify` where the `PipelineRef` is used.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: security-scans
+spec:
+  tasks:
+    - name: scorecards
+      taskRef:
+        name: scorecards
+    - name: codeql
+      taskRef:
+        name: codeql
+---        
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  tasks:
+  - name: git-clone
+    taskRef:
+      name: git-clone
+  - name: security-scans
+    pipeline:
+      pipelineRef:
+        name: security-scans
+  - name: notification
+    taskRef:
+      name: notification
+```
+
+The above example can be modified as such to use `PipelineSpec` instead of `PipelineRef` if the user would prefer to 
+embed the `Pipeline` specification: 
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-scan-notify
+spec:
+  tasks:
+    - name: git-clone
+      taskRef:
+        name: git-clone
+    - name: security-scans
+      pipeline:
+        pipelineSpec:
+          tasks:
+            - name: scorecards
+              taskRef:
+                name: scorecards
+            - name: codeql
+              taskRef:
+                name: codeql
+    - name: notification
+      taskRef:
+        name: notification
+````
```

Comment:
Can you add a section about how this will interact with [taskRunSpecs](https://github.com/tektoncd/pipeline/blob/d300aebecb3849e8727ef88fd4aa0ef88799e8c0/pkg/apis/pipeline/v1beta1/pipelinerun_types.go#L596)?

I think the alternative of inline fields may make more sense, because in that alternative, a PipelineTask holds only the Spec of a Task, Pipeline, or CustomTask. The PipelineRun.TaskRunSpecs would be the right place to put the spec of a PipelineRun IMO.

For example, in the proposed solution, if the user wanted to declare a timeout for their sub-PipelineRun, they would have to do it within the PipelineTask. It would be a bit awkward to have a timeout for the overall Pipeline defined in the PipelineRun, but a timeout for the sub-Pipeline defined in the PipelineTask.